akka-sse/src/main/scala/de/heikoseeberger/akkasse/pattern/Connections.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2015 Heiko Seeberger
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.heikoseeberger.akkasse.pattern
+
+import akka.{ Done, NotUsed }
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{ Flow, Sink, Source }
+import de.heikoseeberger.akkasse.{ EventStreamUnmarshalling, ServerSentEvent }
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.Try
+
+object Connections {
+
+  def sseSource(
+    connection: Flow[HttpRequest, HttpResponse, Future[Http.OutgoingConnection]],
+    request: HttpRequest
+  )(begin: HttpResponse => Unit)(end: Try[Done] => Unit)(
+    implicit
+    ec: ExecutionContext, mat: ActorMaterializer
+  ): Source[ServerSentEvent, NotUsed] =
+    Source
+      .single(request)
+      .via(connection)
+      .alsoTo(
+        Sink.head.mapMaterializedValue(_.foreach { response =>
+          if (response.status.isSuccess())
```

Comment:
Shouldn't we allow the handler to be applied to all responses, i.e. do something specific in case of an unsuccessful response? Hence I think we should remove the `if` clause.